FirebaseSessions/Tests/Unit/Library/LifecycleNotifications.swift

```diff
@@ -36,78 +36,78 @@ import Dispatch
   #endif // os(visionOS)
 #endif // swift(>=5.9)
 
+private func _postBackgroundedNotificationInternal() {
+  let notificationCenter = NotificationCenter.default
+  #if os(iOS) || os(tvOS)
+    notificationCenter.post(name: UIApplication.didEnterBackgroundNotification, object: nil)
+  #elseif os(macOS)
+    notificationCenter.post(name: NSApplication.didResignActiveNotification, object: nil)
+  #elseif os(watchOS)
+    if #available(watchOSApplicationExtension 7.0, *) {
+      notificationCenter.post(
+        name: WKExtension.applicationDidEnterBackgroundNotification,
+        object: nil
+      )
+    }
+  #endif // os(iOS) || os(tvOS)
+
+  // swift(>=5.9) implies Xcode 15+
+  // Need to have this Swift version check to use os(visionOS) macro, VisionOS support.
+  // TODO: Remove this check and add `os(visionOS)` to the `os(iOS) || os(tvOS)` conditional above
+  // when Xcode 15 is the minimum supported by Firebase.
+  #if swift(>=5.9)
+    #if os(visionOS)
+      notificationCenter.post(name: UIApplication.didEnterBackgroundNotification, object: nil)
+    #endif // os(visionOS)
+  #endif // swift(>=5.9)
+}
+
+private func _postForegroundedNotificationInternal() {
+  let notificationCenter = NotificationCenter.default
+  #if os(iOS) || os(tvOS)
+    notificationCenter.post(name: UIApplication.didBecomeActiveNotification, object: nil)
+  #elseif os(macOS)
+    notificationCenter.post(name: NSApplication.didBecomeActiveNotification, object: nil)
+  #elseif os(watchOS)
+    if #available(watchOSApplicationExtension 7.0, *) {
+      notificationCenter.post(
+        name: WKExtension.applicationDidBecomeActiveNotification,
+        object: nil
+      )
+    }
+  #endif // os(iOS) || os(tvOS)
+
+  // swift(>=5.9) implies Xcode 15+
+  // Need to have this Swift version check to use os(visionOS) macro, VisionOS support.
+  // TODO: Remove this check and add `os(visionOS)` to the `os(iOS) || os(tvOS)` conditional above
+  // when Xcode 15 is the minimum supported by Firebase.
+  #if swift(>=5.9)
+    #if os(visionOS)
+      notificationCenter.post(name: UIApplication.didBecomeActiveNotification, object: nil)
+    #endif // os(visionOS)
+  #endif // swift(>=5.9)
+}
+
 extension XCTestCase {
   func postBackgroundedNotification() {
     // On Catalyst, the notifications can only be called on a the main thread
     if Thread.isMainThread {
-      postBackgroundedNotificationInternal()
+      _postBackgroundedNotificationInternal()
```

Comment:
This should really be a `@MainActor` method and then we should use `async` and let Swift take care of the actor checking. Main thread and MainActor are not 1:1: https://www.avanderlee.com/swift/mainactor-dispatch-main-thread/#understanding-why-mainactor-doesnt-always-ensure-main-thread-execution

Reply:
021fb99